src/cpp/include/openvino/genai/tokenizer.hpp

```diff
@@ -241,8 +245,31 @@ class OPENVINO_GENAI_EXPORTS Tokenizer {
     std::shared_ptr<TokenizerImpl> m_pimpl;
 };
 
+/**
+ * @enum PaddingMode
+ * @brief Enum class representing different padding modes for tokenization.
+ *
+ * This enum class defines modes that can be used to pad tokenized sequences. 
+ * IMPORTANT NOTICE: even in TRUNCATE mode the padding is applied to the longest sequence in the batch
+ * since resulting tokenization is stored as a signe ov::Tensor which cannot store irregular/ragged array.
+ * For the same reason also in MAX_LENGTH mode truncation is also applied for sequences longer that max_length,
+ * so that we have the same lengths for each sequence in the batch.
+ *
+ * @var PaddingMode::TRUNCATE
+ * Truncate the sequence to the maximum length specified. (But also LONGEST mode is implicitly applied.)
+ *
+ * @var PaddingMode::LONGEST
+ * Pad the sequence to the length of the longest sequence in the batch. In this mode truncation is switched off.
```

Comment:
> But also LONGEST mode is implicitly applied.

> In this mode truncation is switched off.

So TRUNCATE activates LONGEST which disables TRUNCATE. TRUNCATE switches itself off

Reply:
Looks a bit complicated. I realized that 2 regimes is enough. Please take a look to the latest version.